app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java

```diff
@@ -370,9 +373,12 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
 
             case INTENT_REQUEST_REARRANGE_IMAGE:
                 mImagesUri = data.getStringArrayListExtra(RESULT);
-                if (mImagesUri.size() > 0) {
+                if (!mUnarrangedImagesUri.equals(mImagesUri) && mImagesUri.size() > 0) {
                     showSnackbar(mActivity, R.string.images_rearranged);
-                } else {
+                    mUnarrangedImagesUri.clear();
+                    mUnarrangedImagesUri.addAll(mImagesUri);
+                }
+                if (mImagesUri.size() == 0) {
```

Comment:
I think you can move this `if` block inside the `else` block since `mImagesUri` will only be empty in the `else` block.

Reply:
if we move the if block inside the else block then when images are not arranged the else block will be executed causing the Create pdf button to disable

Reply:
Do we need `mImagesUri.size() > 0` check here?